### ISSUE TITLE
IDISP023 when clearing list within Dispose(bool).

### ISSUE THREAD

**Issue by AraHaan** [DotNetAnalyzers/IDisposableAnalyzers#272]
Inside of the Dispose method at the bottom of the code below warns, however I do not think it should warn.

```cs
namespace AnyNamespaceWorks
{
    using System;
    using System.Collections.Generic;

    public class AnyClassName : IDisposable
    {
        private bool disposedValue;

        // unrelated internal ctor.

        // unrelated event handler.

        // unrelated properties.

        /// <summary>
        /// Gets the plugin name this instance is pointing to.
        /// </summary>
        public string PluginName { get; private protected set; }

        /// <summary>
        /// Gets the current version of the plugin that is pointed to by this instance.
        /// </summary>
        public string CurrentVersion { get; private protected set; }

        /// <summary>
        /// Gets the installed version of the plugin that is pointed to by this instance.
        /// </summary>
        public string InstalledVersion { get; private protected set; }

        /// <summary>
        /// Gets the url to download the files to the plugin from.
        /// </summary>
        public Uri DownloadUrl { get; private protected set; }

        /// <summary>
        /// Gets the files to the plugin to download.
        /// </summary>
        public List<string> DownloadFiles { get; private protected set; }

        // unrelated public static method that calls the internal ctor.

        /// <summary>
        /// Cleans up the resources used by <see cref="AnyClassName"/>.
        /// </summary>
        public void Dispose()
        {
            this.Dispose(true);
            GC.SuppressFinalize(this);
        }

        // unrelated methods snipped.

        private protected virtual void Dispose(bool disposing)
        {
            if (!this.disposedValue && disposing)
            {
                // prevent any leaks from this.
                this.PluginName = null;
                this.CurrentVersion = null;
                this.InstalledVersion = null;
                this.DownloadUrl = null;
                this.DownloadFiles.Clear();
                this.DownloadFiles = null;
                this.disposedValue = true;
            }
        }
    }
}
```

**Comment by JohanLarsson** [comment]
Looks wrong, thanks for the issue!

**Comment by AraHaan** [comment]
@JohanLarsson thanks for the fix, I looked at the appveyor build it the tests started to fail.

**Comment by JohanLarsson** [comment]
Sounds like I dumbed something then, my Visual Studio is flaky so I pushed it without building. Will fix.

**Comment by AraHaan** [comment]
👌, at least you can still salvage the tag since the tag builds failed to publish I think. The tests run before publish right?

**Comment by JohanLarsson** [comment]
Not sure what is wrong from looking at the CI output, bot AppVeyor and DevbOps started failing at the same commit but neither lists any failed tests that I can see.

**Comment by JohanLarsson** [comment]
Found it now

**Comment by AraHaan** [comment]
ye you was right it seems somehow dotnet would exit with exit code 1.

**Comment by AraHaan** [comment]
@JohanLarsson that latest commit fixed it 👌.
